### ISSUE TITLE
Incorrect Parse Error in sparksql / databricks dialect

### ISSUE THREAD

**Issue by anna-azizian** [sqlfluff/sqlfluff#4050]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

```sql
SELECT 'x' AS col
UNION
SELECT 'y' AS col
```
The SQL above doesn't work with the SparkSQL dialect after recent changes (1.4)

### Expected Behaviour

The lint should run without parsing errors.

### Observed Behaviour

I get ` PRS | Line 2, Position 2: Found unparsable section: "UNION\nSELECT 'y' AS col` 

### How to reproduce

Lint the SQL above with `databricks` dialect on `sqlfluff==1.4.0` or `sqlfluff==1.4.1`

### Dialect

SparkSQL/Databricks

### Version

1.4.0 and 1.4.1

### Configuration

Empty

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
